src/MainWindow.vala

```diff
@@ -155,18 +155,33 @@ public class Tasks.MainWindow : Hdy.ApplicationWindow {
 
             listbox.row_selected.connect ((row) => {
                 if (row != null) {
+                    Tasks.ListView? listview;
+
                     if (row is Tasks.SourceRow) {
                         var source = ((Tasks.SourceRow) row).source;
-                        listview.source = source;
-                        Tasks.Application.settings.set_string ("selected-list", source.uid);
+                        var source_uid = source.dup_uid ();
 
-                        listview.add_view (source, "(contains? 'any' '')");
+                        listview = (Tasks.ListView) listview_stack.get_child_by_name (source_uid);
+                        if (listview == null) {
+                            listview = new Tasks.ListView (source);
+                            listview_stack.add_named (listview, source_uid);
+                            listview.add_view (source, "(contains? 'any' '')");
+                        }
 
+                        listview_stack.set_visible_child_name (source_uid);
+                        Tasks.Application.settings.set_string ("selected-list", source_uid);
                         ((SimpleAction) lookup_action (ACTION_DELETE_SELECTED_LIST)).set_enabled (source.removable);
 
                     } else if (row is Tasks.ScheduledRow) {
-                        listview.source = null;
-                        Tasks.Application.settings.set_string ("selected-list", "scheduled");
+                        var scheduled_uid = "scheduled";
+
+                        listview = (Tasks.ListView) listview_stack.get_child_by_name (scheduled_uid);
+                        if (listview == null) {
+                            listview = new Tasks.ListView (null);
+                            listview_stack.add_named (listview, scheduled_uid);
+                        }
+
+                        listview.remove_views ();
```

Comment:
What's this about? If the only time we're doing this now is when we add a scheduled list, shouldn't that just be a part of the way the scheduled list is constructed?

Reply:
Yeah, that's a bit ugly. The reasoning for this is:

We need to create a view (read "query") for each source, so we can display all scheduled tasks across all sources on the scheduled list. But since sources themselves can be created, deleted or deactivated and because I also added the scheduled list to the `Gtk.Stack` it is no more enough doing this just once on constructing the scheduled list.

Therefore I'm removing any potential existing views (queries), when the user sets the visible child of the stack to the scheduled list, then add all views for currently existing and active sources, then displaying the scheduled list.

Other potential solutions for this problem:
- Remove scheduled list from `Gtk.Stack` and fully re-construct it whenever the user switches to the scheduled list
- Keep scheduled list in `Gtk.Stack`, but listen to source crud events (this seemed to cause significant refactoring)